### PR TITLE
[v3.0] Run output plugins last

### PR DESCRIPTION
**Pull request by aleclarson**
This PR contains:
- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [ ] yes (*bugfixes and features will not be merged without tests*)
- [x] no

Breaking Changes?
- [x] yes (*breaking changes will not be merged unless absolutely necessary*)
- [ ] no

List any relevant issue numbers:
- #3845 

### Description

Output plugins will run after input plugins.

Technically a breaking change, but it wasn't documented, so maybe it could be released in a patch version?

### CODE REVIEW

**Comment by lukastaegert** [comment]
Not so sure we can silently do this. It is entirely possible that people rely on that order, though I agree it makes slightly more sense to put output plugins last.

I would see two ways forward:
1. This is marked as a change for the next major release, but this can still take months, and/or
2. We have another look at the proposal to `this.postpone()` certain hooks, though pulling this off properly would likely not be entirely straightforward.

For the time being, I am putting this into the Rollup 3 project.

**Comment by CxRes** [comment]
@lukastaegert Why would the backwards order affect anyone? It would be an extremely contrived case where an input plugin relies on a bundle property (ie objects on the bundle argument in the `generateBundles` hook) which requires an output plugin to have run first. However, having info about all the bundles from input plugins available at the time of processing an output is necessary in many cases.   

It is extremely difficult to believe that on the balance of probabilities that anyone is "on purpose" relying on this property, especially given that it is not even documented anywhere. Even accidental use seems highly unlikely, given how humans think of causality. I really would not like to wait the many months for a stable Rollup 3 to fix this bug!

**Comment by lukastaegert** [comment]
I updated this with the latest release branch, added a test and will merge it into the release branch once green

File: test/form/samples/runs-output-plugins-last/_config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+				name: 'input',
+				renderChunk(code) {
+					return `/* input */\n${code}\n/* input */`;
+				}
+			}
+		],
+		output: {
+			plugins: [
+				{
+					name: 'input',
```
Comment (aleclarson) [inline test/form/samples/runs-output-plugins-last/_config.js:15]: typo i think?

**Comment by lukastaegert** [comment]
This PR has been released as part of rollup@3.0.0-3. Note that this is a pre-release, so to test it, you need to install Rollup via `npm install rollup@3.0.0-3` or `npm install rollup@beta`. It will likely become part of a regular release later.

**Comment by lukastaegert** [comment]
This PR has been released as part of rollup@3.0.0-3. Note that this is a pre-release, so to test it, you need to install Rollup via `npm install rollup@3.0.0-3` or `npm install rollup@beta`. It will likely become part of a regular release later.

**Comment by lukastaegert** [comment]
This PR has been released as part of rollup@3.0.0-4. Note that this is a pre-release, so to test it, you need to install Rollup via `npm install rollup@3.0.0-4` or `npm install rollup@beta`. It will likely become part of a regular release later.

**Pull request merged**